oshi-core/src/main/java/oshi/util/GlobalConfig.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * OSHI (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
+ * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The global configuration utility. See
+ * {@code src/main/resources/default.properties} for default values.
+ * 
+ * @author cilki
+ * @since 4.0.0
+ */
+public final class GlobalConfig {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GlobalConfig.class);
+
+    private static final Properties configuration = new Properties();
+
+    static {
+        try (InputStream in = GlobalConfig.class.getResourceAsStream("/default.properties")) {
+            if (in != null)
+                configuration.load(in);
+        } catch (IOException e) {
+            LOG.warn("Failed to load default configuration");
+        }
+    }
+
+    /**
+     * Get the {@code String} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the given default if not found
+     */
+    public static String get(String key, String def) {
+        return configuration.getProperty(key, def);
+    }
+
+    /**
+     * Get the {@code Integer} property associated with the given key.
```

Comment:
You indicate the boxed type here but return value is primitive (which I like).  Same comment on the `Boolean`.